lib/ui/painting/single_frame_codec.cc

```diff
@@ -108,10 +108,7 @@ Dart_Handle SingleFrameCodec::getNextFrame(Dart_Handle callback_handle) {
 }
 
 size_t SingleFrameCodec::GetAllocationSize() const {
-  const auto& data_size = descriptor_->GetAllocationSize();
-  const auto frame_byte_size =
-      cached_image_ ? cached_image_->GetAllocationSize() : 0;
-  return data_size + frame_byte_size + sizeof(this);
+  return sizeof(*this);
```

Comment:
We shouldn't just use the base implementation here because this object is bigger than `Codec`. We also shouldn't use `sizeof(this)` because that's the size of a pointer, not the size of this class. Argh.

We also shouldn't include the sizes of other objects the VM already knows about here.

Reply:
And in particular, the descriptor gets nulled out once we get a frame, which caused a crash here in the test.